deployment/deploy-data

```diff
@@ -26,29 +25,46 @@ if [ "${ENVIRONMENT}" != "beta" ] && [ "${ENVIRONMENT}" != "production" ] ;then
     exit 1
 fi
 
-# Expects path to release directory as second argument. Removes trailing / if present.
-RELEASEDIRECTORYPATH=$2
-if [ "${RELEASEDIRECTORYPATH: -1}" == "/" ] ;then
-    RELEASEDIRECTORYPATH=${RELEASEDIRECTORYPATH%?}
-fi
+# Expects path to release archive as second argument. Removes trailing / if present.
+RELEASE_ARCHIVE_PATH=$2
 
-RELEASEDIRECTORY=$(basename $RELEASEDIRECTORYPATH)
+RELEASE_DIRECTORY_NAME=$(basename ${RELEASE_ARCHIVE_PATH} .tar.gz)
 REMOTERELEASES=/var/www/backend/${ENVIRONMENT}/django/downloads/releases
 
+# directory to unpack release archive on the deployment host
+REMOTE_STAGING=/home/brca/deploy_staging
+
 # Copies directory to downloads/releases/
-rsync -a --rsync-path='sudo rsync' $RELEASEDIRECTORYPATH ${USER}@${HOST}:$REMOTERELEASES/
+rsync -a -e "ssh -p ${SSH_PORT}" --rsync-path='sudo rsync' $RELEASE_ARCHIVE_PATH ${USER}@${HOST}:${REMOTERELEASES}/${RELEASE_DIRECTORY_NAME}/
 
 # Locations of files on remote server
-DATA=$REMOTERELEASES/$RELEASEDIRECTORY/built_with_change_types.tsv
-REMOVED=$REMOTERELEASES/$RELEASEDIRECTORY/removed.tsv
-METADATA=$REMOTERELEASES/$RELEASEDIRECTORY/version.json
-DIFFJSON=$REMOTERELEASES/$RELEASEDIRECTORY/diff.json
-REPORTS=$REMOTERELEASES/$RELEASEDIRECTORY/reports.tsv
+REMOTE_RELEASE_DIR="${REMOTE_STAGING}/output/release"
+REMOTE_TAR=${REMOTERELEASES}/${RELEASE_DIRECTORY_NAME}/${RELEASE_DIRECTORY_NAME}.tar.gz 
+DATA=${REMOTE_RELEASE_DIR}/built_with_change_types.tsv
+REMOVED=${REMOTE_RELEASE_DIR}/diff/removed.tsv
+METADATA=${REMOTE_RELEASE_DIR}/metadata/version.json
+DIFFJSON=${REMOTE_RELEASE_DIR}/diff/diff.json
+REPORTS=${REMOTE_RELEASE_DIR}/artifacts/reports.tsv
 
-ssh -l${USER} ${HOST} <<-ENDSSH
+ssh -p ${SSH_PORT} -l${USER} ${HOST} <<-ENDSSH
     set -o errexit
-    . /var/www/backend/${ENVIRONMENT}/virtualenv/bin/activate
+
+    # unpacking of tar into a staging dir
+    [ -e ${REMOTE_STAGING} ] && rm -r ${REMOTE_STAGING}    
```

Comment:
am i understanding correctly that we're removing previous release data each time a new release is deployed?

what i've found is that we generally end up with multiple versions of each release scattered across different servers as we debug and fix minor issues in each release. it would be wise of us to store all of the definitive release archives (the ones we actually insert into the database) somewhere for easy access. it would probably also be smart to protect them from deletion/overwriting.

What do you think?